core/src/worker/activities/local_activities.rs

```diff
@@ -531,37 +533,92 @@ impl LocalActivityManager {
                 }
             }
 
-            match status {
-                LocalActivityExecutionResult::Completed(_)
-                | LocalActivityExecutionResult::TimedOut(_)
-                | LocalActivityExecutionResult::Cancelled { .. } => {
-                    // Timeouts are included in this branch since they are not retried
-                    self.complete_notify.notify_one();
-                    LACompleteAction::Report(info)
-                }
-                LocalActivityExecutionResult::Failed(f) => {
-                    if let Some(backoff_dur) = info.la_info.schedule_cmd.retry_policy.should_retry(
+            enum Outcome {
+                FailurePath { backoff: Option<Duration> },
+                JustReport,
+            }
+            macro_rules! calc_backoff {
```

Comment:
Wy do you need a macro for this, seems like overkill.

Reply:
There's no way to simply re-use the code besides this because things would be double-moved.